File: Data Structures & Algorithms/top-k-elements-in-list/submission-1.cpp
class Solution {
public:
    vector<int> topKFrequent(vector<int>& nums, int k) {
        unordered_map<int,int> freq;
        vector<int> ans;
        for(int i:nums){
            freq[i]++;
        }
        vector<vector<int>> bucket(nums.size()+1);

        for(auto j:freq){
            bucket[j.second].push_back(j.first);
        }

        for(int i=bucket.size()-1;i>=0;i--){
            for(int nums:bucket[i]){
                ans.push_back(nums);
                if(ans.size()==k){
                    return ans;
                }
            }
        }
        return nums;
    }
};

File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-1.py
class Solution:
    def findMin(self, nums: List[int]) -> int:
        n=len(nums)
        l=0
        r=n-1
        ans=nums[l]
        while(l<=r):
            if(nums[l]<nums[r]):
                ans=min(ans,nums[l])
                break
            m=l+(r-l)//2
            ans=min(ans,nums[m])
            if nums[m]>=nums[l]:
                l=m+1
            else:
                r=m-1
        return ans
